### PR TITLE
[libgit2] Update to 0.99.0

### PR DESCRIPTION
**Pull request by kevinlul**
**Describe the pull request**

- What does your PR fix? https://github.com/libgit2/libgit2/releases/tag/v0.99.0

- Which triplets are supported/not supported? Have you updated the CI baseline? N/A

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)? Yes

Changes might be breaking but maintainers say it is expected to be API-compatible with the previous version.

### CODE REVIEW

File: ports/libgit2/portfile.cmake [hunk 1]
```diff
@@ -8,8 +8,8 @@ include(vcpkg_common_functions)
 vcpkg_from_github(
```
Comment (LilyWangL) [inline ports/libgit2/portfile.cmake]: `include(vcpkg_common_functions)` is deprecated. Please remove it.

File: ports/libgit2/portfile.cmake [hunk 2]
```diff
@@ -8,8 +8,8 @@ include(vcpkg_common_functions)
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO libgit2/libgit2
-    REF v0.28.4
-    SHA512 b81160608003b25d9b922d259ebbbbf941b6bd5100fa1875497c8cd29de320e292fff568c757a7a85b2b3044ddc1cb92c74dbcb13d630d62ecf9a8559b619d15
+    REF v0.99.0
+    SHA512 e38e18da0e6ed1e5c8198c9eb2c362b21da2d0b9c8bc23309d2f70183549f4b9f23a6db8ce5f1f0f24b373e6427039c2a845b62dd74f91b02cfe8954f961a91b
     HEAD_REF master
 )
```
Comment (LilyWangL) [inline ports/libgit2/portfile.cmake]: Can you please use command `file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)` to handle copyright?

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: ports/libgit2/portfile.cmake [hunk 3]
```diff
@@ -3,13 +3,11 @@ if (VCPKG_CMAKE_SYSTEM_NAME STREQUAL WindowsStore)
     message(FATAL_ERROR "Error: UWP builds are not supported.")
```
Comment (LilyWangL) [inline ports/libgit2/portfile.cmake]: Could you please use command `vcpkg_fail_port_install(ON_TARGET "uwp")` to handle `UWP builds are not supported`?

**Review by LilyWangL — CHANGES_REQUESTED** [review]

**Comment by strega-nil** [comment]
Thanks @kevinlul :)

**Pull request merged**